src/platform/FreeRTOS/GenericThreadStackManagerImpl_FreeRTOS.ipp

```diff
@@ -114,6 +115,24 @@ BaseType_t GenericThreadStackManagerImpl_FreeRTOS<ImplClass>::SignalThreadActivi
     return yieldRequired;
 }
 
+template <class ImplClass>
+void GenericThreadStackManagerImpl_FreeRTOS<ImplClass>::OnJoinerTimer(TimerHandle_t xTimer)
+{
+    GenericThreadStackManagerImpl_FreeRTOS<ImplClass> * self =
+        static_cast<GenericThreadStackManagerImpl_FreeRTOS<ImplClass> *>(pvTimerGetTimerID(xTimer));
+
+    if (xTaskGetTickCount() > self->mJoinerExpire || self->Impl()->IsThreadProvisioned())
+    {
+        ChipLogDetail(DeviceLayer, "Thread joiner timer stopped.");
+        xTimerStop(xTimer, 0);
```

Comment:
should we check return codes in some form of ASSERT ?

Reply:
@andy31415 ?